### PR TITLE
Bento-1345-Local Find

### PR DESCRIPTION
**Pull request by skiran86**
## Description

Bento-1345-Local Find

### CODE REVIEW

File: src/pages/dashboardTab/store/dashboardReducer.js [hunk 1]
```diff
@@ -294,36 +311,116 @@ export function clearSectionSort(groupName) {
...
+      ...getState().autoCompleteSelection,
+    };
+
+    const [
+      caseResponse,
+      fileResponse,
+      sampleResponse,
+      subjectResponse,
+    ] = await Promise.all([
+      getCaseData(variables),
```
Comment (AjayDoddapaneni) [inline src/pages/dashboardTab/store/dashboardReducer.js:410]: We don't need three details you can get the current tab and get the only one.

File: src/pages/dashboardTab/store/dashboardReducer.js [hunk 2]
```diff
@@ -294,36 +311,116 @@ export function clearSectionSort(groupName) {
...
+
+const getFilesDetails = async (variables) => {
+  const result = await client.query({
+    query: CASES_FILE_QUERY,
+    variables: {
+      first: 2,
+      offset: 0,
+      order_by: 'file_id',
+      sort_direction: 'desc',
+      age_at_index: [10, null],
```
Comment (AjayDoddapaneni) [inline src/pages/dashboardTab/store/dashboardReducer.js:368]: Looks like a hardcoded for age_at_index

File: src/pages/dashboardTab/store/dashboardReducer.js [hunk 3]
```diff
@@ -294,36 +311,116 @@ export function clearSectionSort(groupName) {
...
+}
+
+const getSubjectDetails = async (variables) => {
+  const result = await client.query({
+    query: SUBJECT_OVERVIEW_QUERY,
+    variables: {
+      offset: 0,
+      first: 100,
+      sort_direction: 'desc',
+      order_by: 'age_at_index',
```
Comment (AjayDoddapaneni) [inline src/pages/dashboardTab/store/dashboardReducer.js:352]: hard coded filter

File: src/bento/dashboardTabData.js [hunk 4]
```diff
@@ -430,6 +430,192 @@ export const tabIndex = [
   },
 ];
 
+export const DASHBOARD_QUERY_NEW = gql`
```
Comment (AjayDoddapaneni) [inline src/bento/dashboardTabData.js:433]: Can repleace the old queries?

File: src/components/SideBar/SideBarComponents/searchComponet.js [hunk 5]
```diff
@@ -78,11 +81,20 @@ const LocalSearchComponent = ({ classes }, ref) => {
...
+        type: type.replace('Ids', ''),
+        value,
+      });
+      setSideBarToLoading();
+      localSearch(value);
+    }
+  }, [value]);
+
   const onDelete = (title) => () => {
     const newValue = value.filter((v) => v.title !== title);
```
Comment (AjayDoddapaneni) [inline src/components/SideBar/SideBarComponents/searchComponet.js:99]: This needed to be updated aswell

File: src/components/SideBar/SideBarComponents/FacetFilters.js [hunk 6]
```diff
@@ -415,229 +346,90 @@ const FacetPanel = ({ classes }) => {
...
-                                  classes.sortGroup
-                                }
+                          <div
+                            className={classes.sortGroup}
+                          >
+                            <span
+                              className={classes.sortGroupIcon}
+                            >
+                              <Icon
+                                onClick={handleGroupReset(
```
Comment (bremonline) [inline src/components/SideBar/SideBarComponents/FacetFilters.js:377]: Codacy has a fix for the issue: [Unexpected newline after '('.](https://app.codacy.com/gh/CBIIT/bento-frontend/pullRequest?prid=8502143)
```suggestion
                                onClick={handleGroupReset(sideBarItem.datafield, sideBarItem.groupName,
```

File: src/components/SideBar/SideBarComponents/FacetFilters.js [hunk 7]
```diff
@@ -415,229 +346,90 @@ const FacetPanel = ({ classes }) => {
...
+                                  sideBarItem.datafield, sideBarItem.groupName,
+                                )}
```
Comment (bremonline) [inline src/components/SideBar/SideBarComponents/FacetFilters.js:379]: Codacy has a fix for the issue: [Unexpected newline before ')'.](https://app.codacy.com/gh/CBIIT/bento-frontend/pullRequest?prid=8502143)
```suggestion
                                style={{ fontSize: 15 }}
```

File: src/pages/dashboardTab/store/dashboardReducer.js [hunk 8]
```diff
@@ -954,14 +1059,10 @@ const reducers = {
     isDashboardTableLoading: false,
   }),
   TOGGGLE_CHECKBOX_WITH_API: (state, item) => {
-    let updatedCheckboxData1 = updateFilteredAPIDataIntoCheckBoxData(
+    const updatedCheckboxData1 = updateFilteredAPIDataIntoCheckBoxData(
```
Comment (bremonline) [inline src/pages/dashboardTab/store/dashboardReducer.js:1062]: Codacy has a fix for the issue: [Unexpected newline after '('.](https://app.codacy.com/gh/CBIIT/bento-frontend/pullRequest?prid=8502143)
```suggestion
    const updatedCheckboxData1 = updateFilteredAPIDataIntoCheckBoxData(item.data.searchSubjects, facetSearchData,
```

File: src/components/Search/searchAutoFillComponent.js [hunk 9]
```diff
@@ -13,7 +13,7 @@ function searchComponent() {
   const [inputValue, setInputValue] = React.useState('');
   const [options, setOptions] = React.useState([]);
   const loading = open;
-
+
```
Comment (bremonline) [inline src/components/Search/searchAutoFillComponent.js:16]: Codacy has a fix for the issue: [Trailing spaces not allowed.](https://app.codacy.com/gh/CBIIT/bento-frontend/pullRequest?prid=8502143)
```suggestion

```

**Review by wfy1997 — APPROVED** [review]

File: src/components/SideBar/SideBarComponents/FacetFilters.js (hunk 7 above)
Comment (bremonline) [inline src/components/SideBar/SideBarComponents/FacetFilters.js:379]: Codacy has a fix for the issue: [Unexpected newline before ')'.](https://app.codacy.com/gh/CBIIT/bento-frontend/pullRequest?prid=8502143)
```suggestion
                                style={{ fontSize: 15 }}
```

File: src/pages/dashboardTab/store/dashboardReducer.js [hunk 10]
```diff
@@ -954,14 +1023,10 @@ const reducers = {
...
```
Comment (bremonline) [inline src/pages/dashboardTab/store/dashboardReducer.js:1026]: Codacy has a fix for the issue: [Unexpected newline after '('.](https://app.codacy.com/gh/CBIIT/bento-frontend/pullRequest?prid=8502143)
```suggestion
    const updatedCheckboxData1 = updateFilteredAPIDataIntoCheckBoxData(item.data.searchSubjects, facetSearchData,
```

File: src/components/Search/searchAutoFillComponent.js (hunk 9 above)
Comment (bremonline) [inline src/components/Search/searchAutoFillComponent.js:16]: Codacy has a fix for the issue: [Trailing spaces not allowed.](https://app.codacy.com/gh/CBIIT/bento-frontend/pullRequest?prid=8502143)
```suggestion

```

File: src/components/SideBar/SideBarComponents/FacetFilters.js (hunk 6 above)
Comment (bremonline) [inline src/components/SideBar/SideBarComponents/FacetFilters.js:377]: Codacy has a fix for the issue: [Unexpected newline after '('.](https://app.codacy.com/gh/CBIIT/bento-frontend/pullRequest?prid=8502143)
```suggestion
                                onClick={handleGroupReset(sideBarItem.datafield, sideBarItem.groupName,
```

File: src/components/SideBar/SideBarComponents/FacetFilters.js [hunk 11]
```diff
@@ -420,254 +366,230 @@ export const FacetPanel = ({ classes }) => {
...
-                                  classes.sortGroup
-                                }
+                          <div
+                            className={classes.sortGroup}
+                          >
+                            <span
+                              className={classes.sortGroupIcon}
+                            >
+                              <Icon
+                                onClick={handleGroupReset(
```
Comment (bremonline) [inline src/components/SideBar/SideBarComponents/FacetFilters.js:397]: Codacy has a fix for the issue: [Unexpected newline after '('.](https://app.codacy.com/gh/CBIIT/bento-frontend/pullRequest?prid=8502143)
```suggestion
                                onClick={handleGroupReset(sideBarItem.datafield, sideBarItem.groupName,
```

File: src/pages/dashboardTab/store/dashboardReducer.js [hunk 12]
```diff
@@ -969,14 +1055,10 @@ const reducers = {
...
```
Comment (bremonline) [inline src/pages/dashboardTab/store/dashboardReducer.js:1058]: Codacy has a fix for the issue: [Unexpected newline after '('.](https://app.codacy.com/gh/CBIIT/bento-frontend/pullRequest?prid=8502143)
```suggestion
    const updatedCheckboxData1 = updateFilteredAPIDataIntoCheckBoxData(item.data.searchSubjects, facetSearchData,
```

File: src/components/SideBar/SideBarComponents/FacetFilters.js [hunk 13]
```diff
@@ -420,254 +366,230 @@ export const FacetPanel = ({ classes }) => {
...
+                                  sideBarItem.datafield, sideBarItem.groupName,
+                                )}
```
Comment (bremonline) [inline src/components/SideBar/SideBarComponents/FacetFilters.js:399]: Codacy has a fix for the issue: [Unexpected newline before ')'.](https://app.codacy.com/gh/CBIIT/bento-frontend/pullRequest?prid=8502143)
```suggestion
                                style={{ fontSize: 15 }}
```

File: src/bento/dashboardData.js [hunk 14]
```diff
@@ -53,9 +53,6 @@ export const facetSearchData = [
   {
     label: 'File Type', field: 'group', api: 'subjectCountByFileType', apiForFiltering: 'filterSubjectCountByFileType', datafield: 'file_type', section: 'Files', show: true,
   },
-  {
```
Comment (AjayDoddapaneni) [inline src/bento/dashboardData.js:56]: Bring this abck

File: src/bento/dashboardData.js [hunk 15]
```diff
@@ -363,11 +374,6 @@ subjectCountByTissueComposition{
     group
     subjects
 }
-filterSubjectCountByAge{
```
Comment (AjayDoddapaneni) [inline src/bento/dashboardData.js:366]: Bring this back

File: src/pages/dashboardTab/store/dashboardReducer.js (hunk 12 above)
Comment (bremonline) [inline src/pages/dashboardTab/store/dashboardReducer.js:1058]: Codacy has a fix for the issue: [Unexpected newline after '('.](https://app.codacy.com/gh/CBIIT/bento-frontend/pullRequest?prid=8502143)
```suggestion
    const updatedCheckboxData1 = updateFilteredAPIDataIntoCheckBoxData(item.data.searchSubjects, facetSearchData,
```

File: src/components/SideBar/SideBarComponents/FacetFilters.js (hunk 13 above)
Comment (bremonline) [inline src/components/SideBar/SideBarComponents/FacetFilters.js:399]: Codacy has a fix for the issue: [Unexpected newline before ')'.](https://app.codacy.com/gh/CBIIT/bento-frontend/pullRequest?prid=8502143)
```suggestion
                                style={{ fontSize: 15 }}
```

File: src/components/SideBar/SideBarComponents/FacetFilters.js [hunk 16]
```diff
@@ -420,254 +367,230 @@ export const FacetPanel = ({ classes }) => {
...
-                                  classes.sortGroup
-                                }
+                          <div
+                            className={classes.sortGroup}
+                          >
+                            <span
+                              className={classes.sortGroupIcon}
+                            >
+                              <Icon
+                                onClick={handleGroupReset(
```
Comment (bremonline) [inline src/components/SideBar/SideBarComponents/FacetFilters.js:398]: Codacy has a fix for the issue: [Unexpected newline after '('.](https://app.codacy.com/gh/CBIIT/bento-frontend/pullRequest?prid=8502143)
```suggestion
                                onClick={handleGroupReset(sideBarItem.datafield, sideBarItem.groupName,
```

File: src/components/SideBar/SideBarComponents/FacetFilters.js [hunk 17]
```diff
@@ -420,254 +367,230 @@ export const FacetPanel = ({ classes }) => {
...
+                                  sideBarItem.datafield, sideBarItem.groupName,
+                                )}
```
Comment (bremonline) [inline src/components/SideBar/SideBarComponents/FacetFilters.js:400]: Codacy has a fix for the issue: [Unexpected newline before ')'.](https://app.codacy.com/gh/CBIIT/bento-frontend/pullRequest?prid=8502143)
```suggestion
                                style={{ fontSize: 15 }}
```

**Pull request closed without merging**